frontend/src/utils/use_audio.ts

```diff
@@ -50,19 +50,90 @@ export function useAudio({ sources, playbackRate, videoPreview }: UseAudioOption
         'fixed',
         'right-6',
         'bottom-24',
-        'w-[300px]',
-        'h-[171px]',
+        'box-content', // prevent border from messing with aspect ratio
         'bg-black',
         'border-black dark:border-neutral-200',
         'border-2',
         'shadow-brutal',
         'shadow-slate-400 dark:shadow-neutral-600',
         'rounded-lg',
+        'max-w-[calc(100vw-3rem)]',
       );
+
+      const getCursorType = (e: MouseEvent) => {
+        const videoRect = playerElement.getBoundingClientRect();
+        const videoCorner = {
+          x: videoRect.x,
+          y: videoRect.y,
+        };
+        const leftEdge = Math.abs(e.clientX - videoCorner.x) < 10;
+        const topEdge = Math.abs(e.clientY - videoCorner.y) < 10;
+        if (leftEdge && topEdge) {
+          return 'nw-resize';
+        } else if (leftEdge) {
+          return 'w-resize';
+        } else if (topEdge) {
+          return 'n-resize';
+        } else {
+          return 'initial';
+        }
+      };
+
+      let draggingStart: null | {
+        x: number;
+        y: number;
+        initialWidth: number;
+        initialHeight: number;
+        cursorType: 'nw-resize' | 'w-resize' | 'n-resize' | 'initial';
+      } = null;
+      document.addEventListener('pointerdown', (e) => {
+        const videoRect = playerElement.getBoundingClientRect();
+        const cursorType = getCursorType(e);
+        if (cursorType != 'initial') {
+          draggingStart = {
+            x: e.clientX,
+            y: e.clientY,
+            initialWidth: videoRect.width,
+            initialHeight: videoRect.height,
+            cursorType,
+          };
+          e.stopPropagation();
+          e.preventDefault();
+        }
+      });
+      document.addEventListener('pointerup', () => {
+        draggingStart = null;
+      });
+      document.addEventListener('pointermove', (e) => {
```

Comment:
I don't like the idea of registering a global eventlistener for pointermove. This sounds inefficient